### PR TITLE
fix(conformance): exempt tests/integration/ from P017 construction/lifecycle checks

### PR DESCRIPTION
**Pull request by cmgrote**
## Summary
- P017 (`ManualWorkerBootstrap`) fired on every canonical app's integration test harness (`tests/integration/conftest.py` in hello-world/openapi/mysql/metabase-app), since those fixtures build a worker and Temporal client directly in-process to get a handle to submit workflows against and tear down per test.
- `run_dev_combined` has no non-blocking, handle-returning mode that can substitute for that pattern (BLDX-1411), so this was a structural false positive rather than a one-off exception each app should suppress individually.
- Adds `is_integration_test_harness()` and scopes the exemption narrowly to files under `tests/integration/`, and only to the construction-call and lifecycle-call violation classes. The removed-v2-import violation class stays enforced everywhere — those modules don't exist in v3 regardless of context — and a hand-rolled worker anywhere else in the test tree (e.g. a unit test) is still caught.
- Regenerated `prescriptions.md` via the rule-docs generator to reflect the updated P017 rationale/description.

## Test plan
- [x] `uv run pytest packages/conformance/tests -q` (1644 passed, 1 xfailed)
- [x] `uv run python -m conformance.tools.generate_rule_docs --check` (docs up-to-date)
- [x] Verified against the real `atlan-openapi-app/tests/integration/conftest.py` (previously an unsuppressed P017 hit) — now scans clean
- [x] `uv run pre-commit run --all-files`
- [x] `uv run python -m pytest tests/unit -x -q` (5313 passed, 9 skipped)

### CODE REVIEW

**Comment by cmgrote** [comment]
@sdk-review

File: packages/conformance/conformance/suite/checks/entrypoint/_bootstrap_common.py [hunk 1]
```diff
@@ -22,3 +23,22 @@ def is_app_receiver(name: str, origin: str) -> bool:
...
+    this pattern the way it can for app boot code.
+
+    Scoped to the ``tests/integration/`` path shape specifically (both segments
+    present, not just a file/dir *named* "integration") — not to test files in
+    general — so a hand-rolled worker anywhere else in the test tree, e.g. a
+    unit test wrongly reaching for ``create_worker`` instead of a mock, is
+    still caught.
+    """
+    parts = file.replace("\\", "/").split("/")
+    return "tests" in parts and "integration" in parts
```
[the remark this answers is not in the document]
Comment (cmgrote) [reply packages/conformance/conformance/suite/checks/entrypoint/_bootstrap_common.py]: Fixed in 80d9448e — `is_integration_test_harness()` now requires `tests` and `integration` to be *adjacent* path segments (checked via `zip(parts, parts[1:])`), not just present anywhere. Added regression tests for `tests/helpers/integration/...` and `integration/foo/tests/...`, which now correctly still fire.

File: packages/conformance/tests/test_entrypoint.py [hunk 2]
```diff
@@ -203,6 +203,57 @@ def test_silent_on_client_start_workflow(self) -> None:
...
+        assert len(_rule(src, "P017", file="tests/test_integration.py")) == 1
+
+    def test_fires_on_create_worker_under_unit_tests(self) -> None:
+        """A unit test hand-rolling a worker instead of mocking is still caught."""
+        src = (
+            "from application_sdk.execution import create_worker\n"
+            "worker = create_worker(client=client, task_queue='my-app')\n"
+        )
+        assert len(_rule(src, "P017", file="tests/unit/test_something.py")) == 1
+
```
[the remark this answers is not in the document]
Comment (cmgrote) [reply packages/conformance/tests/test_entrypoint.py:255]: Added in 80d9448e: `test_fires_when_integration_segment_not_adjacent_to_tests` and `test_fires_when_tests_segment_not_adjacent_to_integration` cover exactly this case.

File: packages/conformance/conformance/docs/rules/prescriptions.md [hunk 3]
```diff
@@ -575,8 +575,14 @@ and a `run()` / `@entrypoint` method (typed `Input`/`Output`), and set
 run_dev_combined(MyApp, credentials={...}, ...)`. Workers and activities are
 auto-discovered from `AppRegistry` / `TaskRegistry` — there is nothing to wire.
 
+Files under `tests/integration/` are exempt from the construction- call and
```
[the remark this answers is not in the document]
Comment (cmgrote) [reply packages/conformance/conformance/docs/rules/prescriptions.md]: Fixed in 80d9448e — reworded the source lines in `rules/entrypoint.py` so the hyphenated compound word isn't split across a line break, and regenerated `prescriptions.md`.

**Comment by cmgrote** [comment]
## SDK Review: PR #2515 — fix(conformance): exempt tests/integration/ from P017 construction/lifecycle checks

### Verdict: READY TO MERGE

> Narrow, well-targeted fix: adds `is_integration_test_harness()` scoped to adjacent `tests/`→`integration/` path segments, exempts only P017's construction-call and lifecycle-call violation classes (not the dead-v2-import class, which stays enforced everywhere). This treats the root cause identified in the PR body — `run_dev_combined` has no non-blocking, handle-returning mode integration harnesses can use instead (BLDX-1411) — rather than papering over it with per-call-site suppressions. Docs (`prescriptions.md`) were regenerated to match.

---

### Findings

No Critical or Important findings. Reviewed under `review_scope=conformance-only` (all 6 changed files are under `packages/conformance/`) via the conformance-suite lens: detector correctness (AST match precision, false-positive/negative edges), catalog consistency, rule+remediation pairing, and the two CI gates — not SDK Temporal/Dapr assumptions, since this is rule/AST logic, not connector runtime.

Checked and cleared, no issues found:
- **Path-matching precision** — `is_integration_test_harness` requires `tests` immediately followed by `integration` as adjacent path segments (not "both segments present anywhere"). Verified against the adversarial cases the PR's own tests target: `tests/helpers/integration/conftest.py` still fires (sibling tree, not the exempted one), `integration/foo/tests/bar.py` still fires (segments present but not adjacent), and a nested `tests/integration/fixtures/worker.py` is correctly exempt. No gap found.
- **Class 2 (dead v2 imports) stays enforced under the exemption** — confirmed in code (`_is_v2_boot_module` check has no `exempt` guard) and covered by `test_v2_import_still_fires_under_tests_integration`.
- **P018 (server bootstrap) was deliberately left unexempted** — `_server_bootstrap.py` doesn't import `is_integration_test_harness`. Correct: integration harnesses in the canonical apps build a worker+client, not a FastAPI/uvicorn server, so there's no legitimate P018 case to exempt.
- **Catalog/scope drift** — P017's `id`/`scope`/`tier` are unchanged (only `full_description`/rationale text), so `test_catalog.py`'s scope assertions are unaffected. No catalog test gap.
- **Remediation pairing** — no `remediation/**` program references P017 (it's `WARN`-tier, `autofixable=False`, not part of the auto-remediation dispatch), so there's nothing to keep in sync here.
- **Test coverage** — the new public helper `is_integration_test_harness` (exported in `_bootstrap_common.__all__`) has 8 new test cases covering both the positive exemption path and every adjacency/segment edge case that could cause over-matching.
- **Docs sync** — `prescriptions.md` regenerated via the rule-docs generator and verified with `--check` per the PR's test plan; matches the in-code rationale text.

### Holistic Recommendations
None — this is a root-cause fix for a structural false positive (a class of legitimate app-fixture patterns that had no SDK substitute), not a symptom patch. No follow-up needed.

### Strengths
- Two-commit history shows the author self-tightened the match from a looser "both segments present" version to strict adjacency before requesting review — the adversarial edge cases are already covered by tests, not just asserted in prose.
- Kept the dead-import violation class enforced everywhere on purpose, with a test proving it — avoided the common mistake of exempting the whole rule instead of just the classes that need it.
- Verified against a real external consumer (`atlan-openapi-app/tests/integration/conftest.py`) per the test plan, not just synthetic fixtures.

---
**CI:** all passing (no failing checks)
**Review mode:** Interactive Claude Code session (`/sdk-review`), reviewed solo with full PR context loaded (no separate Wave-1 sub-agents or GPT adversarial pass; adversarial is skipped for `conformance-only` scope regardless per policy).
